### ISSUE TITLE
Error with regex variable with multiple `|`s.

### ISSUE THREAD

**Issue by steroidg** [rodjek/puppet-lint#859]
Hi I'm having an issue with Regexp variables. It seems like if I have more than 1 `|` character, puppet-lint fails. From the code below, the first `$kernel_regex` worked fine, but the 2nd and 3rd one failed:

---
puppet-lint version: 2.3.6
ruby version: 2.5.3-p105
platform: x86_64-linux
file path: manifests/params.pp
file contents:
```
# @summary A short summary of the purpose of this class
#
# A description of what this class does
#
# @example
#   include cylance_agent::params
class cylance_agent::params (
) {
  $install_ui_pkg = false

  case $facts['kernel'] {
    'Linux': {
      if $facts['os']['name'] == 'Ubuntu' {
        case $facts['os']['release']['major'] {
          '18.04': {
            $cylance_pkg = 'cylance-protect.1804.x86_64.deb'
            $kernel_regex = /4\.15\.0\-.*-generic/
            $compatible_kernels = [
              '4.15.0-22-generic',
              '4.15.0-23-generic',
              '4.15.0-24-generic',
              '4.15.0-29-generic',
              '4.15.0-30-generic',
              '4.15.0-32-generic',
              '4.15.0-38-generic',
              '4.15.0-43-generic',
              '4.15.0-45-generic',
            ]
          }
          '16.04': {
            $cylance_pkg = 'cylance-protect.1604.x86_64.deb'
            $kernel_regex = /4\.15\.0\-.*-generic|4\.13\.0\-.*-generic|4\.10\.0\-.*-generic|4\.8\.0\-.*-generic|4\.4\.0\-.*-generic/ # lint:ignore:140chars
            $compatible_kernels = [
              '4.4.0-31-generic',
              '4.4.0-81-generic',
              '4.4.0-142-generic',
              '4.8.0-53-generic',
              '4.10.0-21-generic',
              '4.10.0-30-generic',
              '4.10.0-38-generic',
              '4.13.0-31-generic',
              '4.15.0-29-generic',
              '4.15.0-34-generic',
              '4.15.0-38-generic',
              '4.15.0-43-generic',
            ]
          }
          '14.04': {
            $cylance_pkg = 'cylance-protect.1404.x86_64.deb'
            $kernel_regex = /4\.4\.0\-.*-generic|4\.2\.0\-.*-generic|3\.16\.0\-.*-generic|3\.19\.0\-.*-generic|3\.13\.0\-.*-generic|4\.4\.0\-.*-generic|4\.4\.0\-1070/ # lint:ignore:140chars
            $compatible_kernels = [
              '3.13.0-24-generic',
              '3.19.0-80-generic',
              '4.2.0-27-generic',
              '4.4.0-31-generic',
              '4.4.0-78-generic',
              '4.4.0-81-generic',
              '4.4.0-109-generic',
              '4.4.0-1070',
            ]
          }
          default: {
            fail ("os->release->major: ${facts['os']['release']['major']} not supported in ${module_name}")
          }
        }
      } else {
        fail ("os->name: ${facts['os']['name']} not supported in ${module_name}")
      }

      $venue_zone = 'Linux'
    }

    default: {
      fail ("Kernel: ${facts['kernel']} not supported in ${module_name}")
    }
  }
}

```
error:
```
NoMethodError: undefined method `line' for nil:NilClass
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/plugins/check_classes/variable_scope.rb:112:in `block (2 levels) in check'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/plugins/check_classes/variable_scope.rb:57:in `each'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/plugins/check_classes/variable_scope.rb:57:in `block in check'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/plugins/check_classes/variable_scope.rb:40:in `each'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/plugins/check_classes/variable_scope.rb:40:in `check'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checkplugin.rb:21:in `run'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:61:in `block in run'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:58:in `each'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:58:in `run'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint.rb:205:in `run'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:66:in `block in run'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:62:in `each'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:62:in `run'
/opt/puppetlabs/pdk/share/cache/ruby/2.5.0/gems/puppet-lint-2.3.6/bin/puppet-lint:7:in `<top (required)>'
/home/billy/devel/puppet/cylance_agent/bin/puppet-lint:29:in `load'
/home/billy/devel/puppet/cylance_agent/bin/puppet-lint:29:in `<main>'
```

Sorry if this has been reported before, I'm not sure how to search within your list of issues and couldn't find anything similar by looking through some of them.

**Comment by usev6** [comment]
A golfed version of the problematic code:
```
$ cat issue859.pp
class gh::issue859 {
  $regex = /5|6|7/
}
```

If I'm not mistaken the underlying problem is that puppet-lint currently does not expect a regular expression on the right hand side of an assignment. Again if I'm not mistaken this was introduced with Puppet 3.5 -- cmp. https://github.com/puppetlabs/docs-archive/blob/master/puppet/3.5/release_notes.markdown and https://tickets.puppetlabs.com/browse/PUP-525

The line with the assignment to ```$regex``` is currently tokenized as
```
INDENT
VARIABLE
WHITESPACE
EQUALS
WHITESPACE
DIV
NUMBER
PIPE
NUMBER
PIPE
[...]
```

With the below patch (adding ```:EQUALS``` to the hash ```REGEX_PREV_TOKENS```) the line is tokenized as:
```
INDENT
VARIABLE
WHITESPACE
EQUALS
WHITESPACE
REGEX
NEWLINE
```

I'm not sure the patch is good like this, but at least it makes the error go away and it passes all tests:
```
diff --git a/lib/puppet-lint/lexer.rb b/lib/puppet-lint/lexer.rb
index 48cde81..f3475a4 100644
--- a/lib/puppet-lint/lexer.rb
+++ b/lib/puppet-lint/lexer.rb
@@ -99,6 +99,7 @@ class PuppetLint
       :IF      => true,
       :ELSIF   => true,
       :LPAREN  => true,
+      :EQUALS  => true,
     }.freeze
 
     # Internal: An Array of Arrays containing tokens that can be described by
```

**Comment by rodjek** [comment]
Fixed by #882
